File: OBI/neps_academy/exercise_0002.cpp
// TODO: Not working (podium vector iteration)
#include <iostream>
#include <vector>
using namespace std;

int main(){
    // Seu código vai aqui
    int T1, T2, T3;

    vector<int> original = {5, 2, 4, 6, 1, 3};
    vector<int> sorted = {5, 2, 4, 6, 1, 3};
    vector<int> podium[6];

    for (int j = 1; j <= sorted.size(); j++) {
        int chave = sorted[j];
        // Inserir A[j] na sequência ordenada A[1..j-1]
        int i = j - 1;
        while (i >= 0 && sorted[i] > chave) {
            sorted[i + 1] = sorted[i];
            i = i - 1;
        }
        sorted[i + 1] = chave;
    }
    
    for (int x; x <= sorted.size(); x++) {
        for (int y; y <= original.size(); y++) {
            if (sorted[x] == original[y]) {
                int c = 0;
                podium[c].push_back(y);
                c = c + 1;
            }
        }
    }

    for (int x : sorted) cout << x << " ";
    cout << "\n";
    for (int x : original) cout << x << " ";
    cout << "\n";
    for (int x : podium[0]) cout << x << " ";
    cout << "\n";
    
    return 0;
}

File: OBI/neps_academy/exercise_0002.py
a = int(input())
b = int(input())
c = int(input())

A = [a, b, c]
O = A[:]
podium = []

for j in range(1, len(A)):
    chave = A[j]
    # Inserir A[j] na sequência ordenada A[1..j-1]
    i = j - 1
    while i >= 0 and A[i] > chave:
        A[i + 1] = A[i]
        i = i - 1
    
    A[i + 1] = chave

for x in range(0, len(A)):
    for y in range(0, len(O)):
        if A[x] == O[y]:
            podium.append(y + 1)

for z in podium:
    print(z)